internal/mempool/mempool_test.go

```diff
@@ -3189,3 +3189,63 @@ func TestMiningView(t *testing.T) {
 			"even though it was removed from the mempool.")
 	}
 }
+
+// TestAncestorTrackingLimits ensures that ancestor tracking is
+// disabled for transactions that have too many ancestors.
+func TestAncestorTrackingLimits(t *testing.T) {
+	harness, spendableOuts, err := newPoolHarness(chaincfg.MainNetParams())
+	if err != nil {
+		t.Fatalf("unable to create test pool: %v", err)
+	}
+
+	// Add a chain of transactions to the mempool.
+	var allTxns []*dcrutil.Tx
+	prevSpendableOut := spendableOuts[0]
+	for i := 0; i < ancestorTrackingLimit+1; i++ {
```

Comment:
I believe this needs to be at least `ancestorTrackingLimit+2` (or `i <= ancestorTrackingLimit+1`), otherwise you aren't going to cross the `ancestorTrackingLimit` threshold.

Reply:
You're right, this has been updated. Thanks.